nncf/experimental/torch2/function_hook/wrapper.py

```diff
@@ -31,70 +31,76 @@
 class ForwardWithHooks:
     """Class to wrap forward function of nn.Module, to forward function of the model with enabled FunctionHookMode"""
 
-    __slots__ = "_func", "__dict__", "__weakref__"
-    _func: Callable[..., Any]
-
-    def __new__(cls, orig_forward: Callable[..., Any]) -> ForwardWithHooks:
-        if not callable(orig_forward):
-            msg = "the first argument must be callable"
-            raise TypeError(msg)
+    __slots__ = "_orig_forward", "_model", "__dict__", "__weakref__"
+    _orig_forward: Callable[..., Any]
+    _model: nn.Module
```

Comment:
Could the storing a reference to the module in hooks potentially result in issues with the removal of the model?

Reply:
I would suggest checking the case with the model's coping.

Reply:
The reference to module contains in `orig_forward` too under `__self__` attributes (in case of bound method) and collected in arg[0] in patched forward by `functools.partial` (for accelerate).
This changes only about getting model only ones on wrapping, on develop model gets from `orig_forward` each forward.

Reply:
Thanks for the answer.